apps/dev/src/commands/feeds/read/adfs.ts

```diff
@@ -9,9 +9,43 @@ import {
 import { renderTui, drawTable } from '@blocksense/base-utils/tty';
 import { listEvmNetworks, readEvmDeployment } from '@blocksense/config-types';
 import { AggregatedDataFeedStoreConsumer } from '@blocksense/contracts/viem';
+import { formatTimestamp } from '../../utils';
 
 const availableNetworks = await listEvmNetworks();
 
+export const parseADFSData = (hexData: string) => {
+  const cleanHex = hexData.startsWith('0x') ? hexData.slice(2) : hexData;
+  const valuePart = cleanHex.slice(0, 48);
+  const valueHex = '0x' + valuePart.padEnd(64, '0');
+  let value: bigint = 0n;
+  try {
+    value = BigInt(valueHex);
+  } catch {
+    // ignore
+  }
```

Comment:
As we discussed, we only use this function for numerical data. Therefore if there is problem with the parsing, throwing here is fine as this might show incorrect usage.
```suggestion
  const value = BigInt(valueHex);
```